### ISSUE TITLE
missing sandbox path throws: terminate called without an active exception

### ISSUE THREAD

**Issue by milahu** [NixOS/nix#5983]
**Describe the bug**

nix throws `terminate called without an active exception`
when a sandbox path does not exist

**Steps To Reproduce**

add this to `/etc/nixos/configuration.nix`

```nix
{
  #nix.package = pkgs.nixUnstable;
  nix.extraOptions = ''
    extra-sandbox-paths = /does/not/exist
  '';
```

run `sudo nixos-rebuild switch`

run `nix-build`

```
nix-build -E 'with import <nixpkgs> { }; callPackage ({pkgs}: pkgs.stdenv.mkDerivation{ name="x"; }) { }'
```

**Expected behavior**

fatal error (i guess)

**`nix-env --version` output**

nix-env (Nix) 2.5pre20211007_844dd90

**Comment by thufschmitt** [comment]
I can reproduce it on 2.5pre20211007_844dd90, but looks like it’s already fixed on 2.6.0:

```console
$ nix --version      
nix (Nix) 2.6.0
$ nix-build -E 'with import <nixpkgs> { }; callPackage ({pkgs}: pkgs.stdenv.mkDerivation{ name="x"; }) { }' --option extra-sandbox-paths /foo/bar --store $PWD   
warning: Nix search path entry '/home/regnat/Config/overlays' does not exist, ignoring
this derivation will be built:
  /nix/store/95gk4zydvxdjvlhiia3q40v297rcdanz-x.drv
error: getting attributes of path '/foo/bar': No such file or directory
(use '--show-trace' to show detailed location information)
```

**Comment by milahu** [comment]
aah, thanks
